app/code/Magento/OpenSearch/SearchAdapter/Adapter.php

```diff
@@ -111,6 +122,7 @@ public function query(RequestInterface $request) : QueryResponse
             $rawResponse = $client->query($query);
         } catch (\Exception $e) {
             $this->logger->critical($e);
+            $this->response->setNoCacheHeaders();
             // return empty search result in case an exception is thrown from OpenSearch
```

Comment:
Maybe make sense to rethrow exception here?

Reply:
I think it's not a good idea

Reply:
Can you please  explain why?

Reply:
@kandy I'd also vote for @rogerdz solution:
Certainly, the proposed change will prevent the display of products in case of error; however, it significantly preserves the user experience compared to displaying a "500 SERVER ERROR". With this approach, critical features such as menus and the shopping cart will remain accessible to visitors, avoiding a complete disruption of site functionality.